api/registrations/views.py

```diff
@@ -93,21 +95,21 @@ class RegistrationMixin(NodeMixin):
     serializer_class = RegistrationSerializer
     node_lookup_url_kwarg = 'node_id'
 
-    def get_node(self, check_object_permissions=True):
-        node = get_object_or_error(
-            AbstractNode,
-            self.kwargs[self.node_lookup_url_kwarg],
-            self.request,
-            display_name='node',
+    def get_node(self, check_object_permissions=True, **annotations):
+        guid = self.kwargs[self.node_lookup_url_kwarg]
+        node = Registration.objects.filter(guids___id=guid).annotate(**annotations)
 
-        )
-        # Nodes that are folders/collections are treated as a separate resource, so if the client
-        # requests a collection through a node endpoint, we return a 404
-        if node.is_collection or not node.is_registration:
+        try:
+            node = node.get()
+        except Registration.DoesNotExist:
             raise NotFound
-        # May raise a permission denied
+
+        if node.deleted:
+            raise Gone
```

Comment:
The main thing lost by not using `get_object_or_error` is a custom error message here -- `display_name='node'` led to `Gone(detail='The requested node is no longer available.')` in this case.

It'd be nice to be able to pass annotations to that function, but at minimum we shouldn't lose the error message.